elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -78,7 +108,8 @@ public FilterExpression visitPermissionClass(ExpressionParser.PermissionClassCon
             }
             return ((FilterExpressionCheck) check).getFilterExpression(requestScope);
         } else if (UserCheck.class.isAssignableFrom(check.getClass())) {
-            return USER_CHECK_EXPRESSION;
+            boolean userCheckResult = check.ok(requestScope.getUser());
+            return new UserCheckExpression(userCheckResult);
```

Comment:
if userCheckResult is true, return NO_EVALUATION_EXPRESSION.  Otherwise return USER_CHECK_EXPRESSION.